src/transport.js

```diff
@@ -80,6 +80,9 @@ module.exports = function (swarm) {
       const createListeners = multiaddrs.map((ma) => {
         return (cb) => {
           const listener = transport.createListener(handler)
+
+          listener.once('error', cb)
+
```

Comment:
And I don't know if we should do that. Shouldn't the consumers know about such errors and be able to react to them as they see appropriate?

The thing is, these errors are unhandled atm which makes it cause an "unhandled exception" in the higher layer, ie. it needs to be caught on a global level (as opposed to block level). The correct way is to make sure there are no unhandled exceptions.